x-pack/plugins/endpoint/public/applications/endpoint/store/policy_details/reducer.ts

```diff
@@ -93,10 +95,19 @@ export const policyDetailsReducer: Reducer<PolicyDetailsState, AppAction> = (
       return state;
     }
     const newState = { ...state, policyItem: { ...state.policyItem } };
-    const newPolicy: any = { ...fullPolicy(state) };
+    const newPolicy: PolicyConfig = { ...fullPolicy(state) };
+
+    /**
+     * This is directly changing redux state because `policyItem.inputs` was copied over and not cloned.
+     */
+    // @ts-ignore
     newState.policyItem.inputs[0].config.policy.value = newPolicy;
 
     Object.entries(action.payload.policyConfig).forEach(([section, newSettings]) => {
+      /**
+       * this is not safe because `action.payload.policyConfig` may have excess keys
+       */
+      // @ts-ignore
```

Comment:
suggestion: store the `action.payload` on state plain and simple
```
return {
  userModifiedPolicyConfig: action.payload
}
```

Then in the middleware when you need to make a request, create a new object that is a combination of the policy (as it was received from the server) and the user modified parts.

Reply:
My initial goal was actually the opposite - ensure the `state.policyItem` is alway a full representation of the datasource/policy and create selectors that return back a sub-section of that for use in the UI.
I'm good with changing the approach if that makes things easier to work with from a TS standpoint.

Reply:
gotcha. My suggestion wasn't really to get around any TS issues. I try to model redux state as a normalized representation of the application state. In this case, the form dispatches a series of actions, each including the entire state of the form. That's why I recommended making the state be simply the payload of the action. This also keeps writing to the redux store cheap, which is not a primary goal, but doesn't hurt.

I would say that merging the form state with other metadata to make a request to another plugin would be the concern of the data layer, that's why I recommend doing it in the middleware. 

Anyway, just my 2c